utils/node_local_test.py

```diff
@@ -276,31 +295,96 @@ def start(self):
+                             '--xml=yes',
+                             '--xml-file=dnt_server.%p.memcheck.xml',
+                             '--num-callers=2',
+                             '--leak-check=no',
+                             '--keep-stacktraces=none',
+                             '--undef-value-errors=no']
+            self._io_server_dir = tempfile.TemporaryDirectory(prefix='dnt_io_')
+
+            fd = open(os.path.join(self._io_server_dir.name,
+                                   'daos_io_server'), 'w')
+            fd.write('#!/bin/sh\n')
+            fd.write('export PATH=$REAL_PATH\n')
+            fd.write('exec valgrind {} daos_io_server "$@"\n'.format(
+                ' '.join(valgrind_args)))
+            fd.close()
+
+            os.chmod(os.path.join(self._io_server_dir.name, 'daos_io_server'),
+                     stat.S_IXUSR | stat.S_IRUSR)
+
+            server_env['REAL_PATH'] = '{}:{}'.format(
+                os.path.join(self.conf['PREFIX'], 'bin'), server_env['PATH'])
+            server_env['PATH'] = '{}:{}'.format(self._io_server_dir.name,
+                                                server_env['PATH'])
+
+        cmd = [daos_server, '--config={}'.format(self._yaml_file.name),
                'start', '-t' '4', '--insecure', '-d', self.agent_dir,
                '--recreate-superblocks']
 
-        server_env = get_base_env()
         server_env['DAOS_DISABLE_REQ_FWD'] = '1'
         self._sp = subprocess.Popen(cmd, env=server_env)
 
         agent_config = os.path.join(self_dir, 'nlt_agent.yaml')
 
         agent_bin = os.path.join(self.conf['PREFIX'], 'bin', 'daos_agent')
 
+        agent_log_file = tempfile.NamedTemporaryFile(prefix='dnt_agent_',
+                                                     suffix='.log',
+                                                     delete=False)
+
         self._agent = subprocess.Popen([agent_bin,
                                         '--config-path', agent_config,
                                         '--insecure',
                                         '--debug',
                                         '--runtime_dir', self.agent_dir,
-                                        '--logfile', '/tmp/dnt_agent.log'],
+                                        '--logfile', agent_log_file.name],
                                        env=os.environ.copy())
         self.conf.agent_dir = self.agent_dir
-        time.sleep(2)
         self.running = True
 
+        # Use dmg to block until the server is ready to respond to requests.
+        start = time.time()
+        while True:
+            time.sleep(0.5)
+            rc = self.run_dmg(['system', 'query'])
+            ready = False
+            if rc.returncode == 0:
+                for line in rc.stdout.decode('utf-8').splitlines():
+                    if line.startswith('status'):
+                        if 'Ready' in line:
+                            ready = True
```

Comment:
I was going to suggest that you could use the `--json` output option instead of doing string parsing, but I just noticed that we're not using the stringified representation, so it's probably not any better.